### PR TITLE
Responsive pagination

### PR DESCRIPTION
**Pull request by alisey**

### CODE REVIEW

File: packages/wix-ui-core/src/components/Pagination/PageStrip.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    public componentDidUpdate() {
+        this.updateLayoutIfNeeded();
+    }
+
+    public render() {
+        return (
+            <div
+                data-hook="page-strip"
+                id={this.props.id ? this.props.id + 'pageStrip' : null}
+                className={this.props.classes.pageStrip}
```
Comment (amiryonatan) [inline packages/wix-ui-core/src/components/Pagination/PageStrip.tsx]: const {id, classes} = this.props;

File: packages/wix-ui-core/src/components/Pagination/Pagination.protractor.driver.ts [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
   element: () => component,
+
   /** Returns x & y coordinates for the element found with data-hook */
   getElementLocation: (dataHook): Promise<{x: number, y: number}> => component.$(`[data-hook="${dataHook}"]`).getLocation(),
-  /** Returns width &N height for the element found with data-hook */
-  getElementSize: (dataHook): Promise<{width: number, height: number}> => component.$(`[data-hook="${dataHook}"]`).getSize()
+
+  /** Returns width & height for the element found with data-hook */
+  getElementSize: (dataHook): Promise<{width: number, height: number}> => component.$(`[data-hook="${dataHook}"]`).getSize(),
+
```
Comment (JoMarton) [inline packages/wix-ui-core/src/components/Pagination/Pagination.protractor.driver.ts]: please add comments for the storybook autodocs

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -109,109 +122,83 @@ class Pagination extends React.Component<PaginationProps, PaginationState> {
...
     }
   }
 
-  private pageInputCommit = (): void => {
-    const page = Number(this.state.pageInputValue);
-    if (page && page !== this.currentPage) {
-      this.onChange(this.makePageNumberValid(page, this.props.totalPages));
+  private handleNavButtonKeyDown = (event: React.KeyboardEvent<Element>, page: number): void => {
+    // Enter or Space
+    if (event.keyCode === 13 || event.keyCode === 32) {
```
Comment (JoMarton) [inline packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: Please write tests for this

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -33,7 +34,8 @@ export interface PaginationProps {
   totalPages: number;
   currentPage?: number;
   // props
-  onChange?: (event: {page: string}) => void;
+  pageUrl?: (pageNumber: number) => string;
+  onChange?: ({event, page}: {event: React.SyntheticEvent<Element>, page: number}) => void;
```
Comment (JoMarton) [inline packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: although lint doesn't complain, my IDE really doesn't like this syntax. consider using: ```(e: {event: React.SyntheticEvent<Element>, page: number}) => void;```

File: packages/wix-ui-core/src/components/Pagination/theme.ts [hunk 5]
```diff
@@ -36,8 +36,9 @@ export const core: PaginationTheme = {
   },
   pageStrip: {
     display: 'flex',
-    overflow: 'hidden',
+    order: 3,
```
Comment (JoMarton) [inline packages/wix-ui-core/src/components/Pagination/theme.ts]: currently "order" is inline style for the other elements. if this is in the theme then all should be in theme

**Pull request merged**